tests/descriptor_tests/test_io_descriptors.py

```diff
@@ -22,6 +22,146 @@ class TestIODescriptors(TankTestBase):
+            releases.append(
+                sgtk.descriptor.create_descriptor(
+                    sg,
+                    sgtk.descriptor.Descriptor.APP, {
+                        "type": "app_store",
+                        "version": release,
+                        "name": "tk-bundle"
+                    },
+                    bundle_cache_root_override=root,
+                )
+            )
+            # check the uri, we shouldn't have any label param in it
+            # sgtk:descriptor:app_store?version=v2.1.2&name=tk-bundle
+            self.assertFalse("label" in releases[-1].get_uri())
+            # Create a master release, where master is explicit
+            releases.append(
+                sgtk.descriptor.create_descriptor(
+                    sg,
+                    sgtk.descriptor.Descriptor.APP, {
+                        "type": "app_store",
+                        "version": "%s-master" % release,
+                        "label": "master",
+                        "name": "tk-bundle"
+                    },
+                    bundle_cache_root_override=root,
+                )
+            )
+            self.assertTrue("&label=master" in releases[-1].get_uri())
+            # Create a branch release
+            releases.append(
+                sgtk.descriptor.create_descriptor(
+                    sg,
+                    sgtk.descriptor.Descriptor.APP, {
+                        "type": "app_store",
+                        "version": "%s-topic" % release,
+                        "label": "topic",
+                        "name": "tk-bundle"
+                    },
+                    bundle_cache_root_override=root,
+                )
+            )
+            # check the uri, we should have label=topic in it
+            # e.g. sgtk:descriptor:app_store?version=v2.1.2.topic&name=tk-bundle&label=topic
+            self.assertTrue("&label=topic" in releases[-1].get_uri())
+
+        sg_bundle_data = {
+            "id": 666,
+            "sg_system_name": "tk-bundle",
+            "sg_status_list": "ip",
+            "sg_deprecation_message": "",
+        }
+        # Create dummy app payload
+        for i, release in enumerate(releases):
+            app_path = os.path.join(root, "app_store", "tk-bundle", release.version)
+            path = os.path.join(app_path, "info.yml")
+            if not os.path.exists(app_path):
+                os.makedirs(app_path)
+            fh = open(path, "wt")
+            fh.write("test data\n")
+            fh.close()
+
+            # Generate cached meta data
+            sg_data_for_version = {
+                "id": i,
+                "code": release.version,
+                "sg_status_list": "alpha",
+                "description": "",
+                "tag_list": [None, ["master"], ["topic"]][i%3],
+                "sg_detailed_release_notes": "",
+                "sg_documentation": "",
+                "sg_branch": [None, "master", "topic"][i%3],
```

Comment:
It is indeed never used, but could, at the very least to be displayed to the user. I felt it was important to store this information, even if it is not used right now?